### ISSUE TITLE
Getting Error in Exhaustive feature matching

### ISSUE THREAD

**Issue by wvendetta9041** [colmap/colmap#1485]
==== running: colmap exhaustive_matcher --SiftMatching.guided_matching=true --database_path colmap.db

==============================================================================
Exhaustive feature matching
==============================================================================

Shader not supported by your hardware!
ERROR: SiftGPU not fully supported
ERROR: SiftGPU not fully supported

**Comment by wvendetta9041** [comment]
Could anyone help on this, please?

**Comment by tsattler** [comment]
Looks like your GPU does not support the shader that runs feature matching. You could switch to feature matching on the GPU (which will be significantly slower though).

**Comment by wvendetta9041** [comment]
Thanks for your answer!, do you mean switch to CPU, right?

On the other hand, should it be related to the fact that colmap does not recognize correctly CUDA???

**Comment by tsattler** [comment]
Yes, set the `use_gpu` flag to 0.

Since the message says that `Shader not supported by your hardware!`, it looks to me that Colmap is trying to run the OpenGL version of SIFT. Are you using a version of Colmap that was compiled with CUDA? Do you have CUDA installed?

**Comment by wvendetta9041** [comment]
Yes, in fact, I already solved it, I had a colmap version, installed from debian repo, but it was not compiled with CUDA, and I just did.

Thanks for answering!

**Comment by wvendetta9041** [comment]
Now I am having this issue:

22:32:18 INFO     Loading NeRF dataset from
22:32:18 INFO       ./transforms.json
22:32:18 SUCCESS  Loaded 10 images of size 4032x3024 after 0s
22:32:18 INFO       cam_aabb=[min=[1.64166,0.650583,-0.0438667], max=[1.84194,0.747907,1.05516]]
22:32:19 INFO     Loading network config from: /home/gmatheu/instant-ngp/configs/nerf/base.json
22:32:19 INFO     GridEncoding:  Nmin=16 b=1.66248 F=2 T=2^19 L=16
Warning: FullyFusedMLP is not supported for the selected architecture 37. Falling back to CutlassMLP. For maximum performance, raise the target GPU architecture to 75+.
Warning: FullyFusedMLP is not supported for the selected architecture 37. Falling back to CutlassMLP. For maximum performance, raise the target GPU architecture to 75+.
Warning: FullyFusedMLP is not supported for the selected architecture 37. Falling back to CutlassMLP. For maximum performance, raise the target GPU architecture to 75+.
22:32:19 INFO     Density model: 3--[HashGrid]-->32--[FullyFusedMLP(neurons=64,layers=3)]-->1
22:32:19 INFO     Color model:   3--[Composite]-->16+16--[FullyFusedMLP(neurons=64,layers=4)]-->3
22:32:19 INFO       total_encoding_params=13623184 total_network_params=9728
Training:   0%|                                                                                                                                                                | 0/11000 [00:00<?, ?step/s]
Got cutlass error: Error Internal at: 346
Could not free memory: /home/gmatheu/instant-ngp/dependencies/tiny-cuda-nn/include/tiny-cuda-nn/gpu_memory.h:451 cudaDeviceSynchronize() failed with error operation not permitted when stream is capturing

How can I solve it?

**Comment by tsattler** [comment]
This error is not related to Colmap. I'd recommend to ask in the instant-ngp github project.

**Comment by sarveshkoyande** [comment]
@wvendetta9041 can you please elaborate how you solved the earlier problem, im a begineer. Did you use use_gpu flag? If so how?
Or did you compile colmap with cuda, if so how?
I am using google colab to run this code.

**Comment by csm-kr** [comment]
@sarveshkoyande According to https://colmap.github.io/cli.html, '--SiftMatching.use_gpu 0' options make colmap use CPU-based feature extraction and matching.

**Comment by Hezhexi2002** [comment]
> @wvendetta9041 can you please elaborate how you solved the earlier problem, im a begineer. Did you use use_gpu flag? If so how? Or did you compile colmap with cuda, if so how? I am using google colab to run this code.

You can follow the official doc from colmap to compile it with cuda surpport:https://colmap.github.io/install.html#build-from-source,just make sure you have set the CMAKE_CUDA_ARCHITECTURES to “native” in the CMakelists.txt like below if you  have manually installed latest CUDA from NVIDIA’s homepage：
![image](https://user-images.githubusercontent.com/53631206/233101719-5702fe0d-a2c7-47f1-8d3d-58fadd06491b.png)

**Comment by KnightOfTheMoonlight** [comment]
> @sarveshkoyande According to https://colmap.github.io/cli.html, '--SiftMatching.use_gpu 0' options make colmap use CPU-based feature extraction and matching.

Thanks! You save my night!

**Comment by LingyunWang** [comment]
Yes, You need to compile your COLMAP with CUDA. I met the same problem.COLMAP use SIFTGPU to accelerate, which is based on CUDA/OpenGL
